File: problems/3010/solution.rb
# @return {Integer}
def minimum_cost(nums)
  first = second = Float::INFINITY
  nums.drop(1).each do |x|
    if x < first
      first, second = x, first
    elsif x < second
      second = x
    end
  end
  nums[0] + first + second
end

File: problems/3010/solution.ts
function minimumCost(nums: number[]): number {
    let first = Infinity, second = Infinity;
    for (let i = 1; i < nums.length; ++i) {
        const x = nums[i];
        if (x < first) { second = first; first = x; }
        else if (x < second) second = x;
    }
    return nums[0] + first + second;
}
